### ISSUE TITLE
2 cluster trained when small number of data entry

### ISSUE THREAD

**Issue by humblemat810** [ddangelov/Top2Vec#198]
I used your example yahoo to test
but applied a filter that I only use data that has 100 or less space " " (i.e. a way to specify small docs)
i applied this to the readme.md example

data = [doc for doc in data if len(doc.split(" ")) <= 100]
i got around 10000 docs left
and only 2 cluster was found

**Comment by humblemat810** [comment]
the code attached:

import time, pathlib
print(time.ctime())
threshold = 100

import numpy as np

from top2vec import Top2Vec
from sklearn.datasets import fetch_20newsgroups

newsgroups = fetch_20newsgroups(subset='all', remove=('headers', 'footers', 'quotes'))
word_count = np.array([len(doc.split(' ')) for doc in newsgroups.data])
filtered_doc = []
filtered_target = []
for i, cnt in enumerate(word_count):
    if cnt < threshold:
        filtered_doc.append(newsgroups.data[i])
        filtered_target.append(newsgroups.data[i])
        pass

model = Top2Vec(documents=filtered_doc, speed="learn", workers=4)
model_name = pathlib.Path("./top2vecNewsgroup_100_space_max")
model.save(model_name)
print(time.ctime())


# %%


model = Top2Vec.load(model_name)


model.get_num_topics()

![image](https://user-images.githubusercontent.com/67593116/129194788-655bedd8-c8c0-41a0-a6ea-4903dd579fae.png)

**Comment by humblemat810** [comment]
```python
import time, pathlib
print(time.ctime())
threshold = 100

import numpy as np

from top2vec import Top2Vec
from sklearn.datasets import fetch_20newsgroups

newsgroups = fetch_20newsgroups(subset='all', remove=('headers', 'footers', 'quotes'))
word_count = np.array([len(doc.split(' ')) for doc in newsgroups.data])
filtered_doc = []
filtered_target = []
for i, cnt in enumerate(word_count):
    if cnt < threshold:
        filtered_doc.append(newsgroups.data[i])
        filtered_target.append(newsgroups.data[i])
        pass

model = Top2Vec(documents=filtered_doc, speed="learn", workers=4)
model_name = pathlib.Path("./top2vecNewsgroup_100_space_max")
model.save(model_name)
print(time.ctime())


# %%


model = Top2Vec.load(model_name)


model.get_num_topics()`
```

**Comment by humblemat810** [comment]
Currently passed parameter 
hdbscan_args = {'min_cluster_size': 5,
                            'metric': 'euclidean',
                            'cluster_selection_method': 'leaf'}
resulted in 200 up clusters, maybe able to solve the problem when the number of examples is smaller than the answers dataset

**Comment by ddangelov** [comment]
Using `doc2vec` as the embedding model for small datasets may be suboptimal. Try using the `universal-sentence-encoder`.

**Comment by humblemat810** [comment]
The dataset is of the same order of magnitude of the Yahoo answer dataset as in your paper
10621 out of 18846 has been used, but still
switching to universal-sentence-encoder alone cannot solve the problem
still has 2 clusters only
It seems unconvincing that having more than half of the full dataset could have detrimental results that cannot corrected by using universal-sentence-encoder

just a quick summary:

1. universal sentence encoder, for this dataset, does not help
2. however, using "leaf" for hdbscan option, is better.
3. No matter what encoder used, the hdbscan is of higher importance
hdbscan_args = {'min_cluster_size': 15,
                            'metric': 'euclidean',
                            'cluster_selection_method': 'leaf'}

**Extended issue:**
but another question, does it mean that 'leaf' would lead to the loss of topic stability upon different runs?
